test/unit/lodash.spec.js

```diff
@@ -0,0 +1,475 @@
+'use strict';
+
+var chai = require('chai');
+var expect = chai.expect;
+var _ = require('lodash');
+var datasets = require('../../app/datasets');
+// var sinon = require('sinon');
+
+describe('Lodash training', function() {
+    var params;
+
+    // ----------------------- OBJECTS ----------------------- //
+
+    describe('functions', function() {
+        beforeEach(function() {
+            params = datasets.functions();
+        });
+        it('should match types of passing elements', function() {
+            var element = params[0] instanceof Object;
+            expect(element).to.eql(true);
+            expect(params).to.have.length(1);
+        });
+        it('should contain only one function', function() {
+            var obj = params[0];
+            var i = 0;
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    if (obj[key] instanceof Function) {
+                        i++;
+                    }
+                }
+            }
+            expect(i).to.eql(1);
+        });
+        it('should return array property names of functions', function() {
+            expect(_.functions.apply(_, params)).to.eql(['x']);
+        });
+    });
+
+    describe('functionsIn', function() {
+        beforeEach(function() {
+            params = datasets.functionsIn();
+        });
+        it('should match types of passing elements', function() {
+            var element = params[0] instanceof Object;
+            expect(element).to.eql(true);
+            expect(params).to.have.length(1);
+        });
+        it('should not contain proper11111111111111111111111', function() {
+            var obj = params[0];
+            for (var key in obj) {
+                if (obj.hasOwnProperty(key)) {
+                    // co tu wpisac ...
+                }
+            }
+        });
+        it('should return array property names of functions', function() {
+            expect(_.functionsIn.apply(_, params)).to.eql(['x', 'z']);
+        });
+    });
+
+    describe('pick', function() {
+        beforeEach(function() {
+            params = datasets.pick();
+        });
+        it('should match types of passing elements', function() {
+            var element = params[0] instanceof Object;
+            var element1 = params[1] instanceof Object;
+            expect(element).to.eql(true);
+            expect(element1).to.eql(true);
+            expect(params.length).to.eql(2);
+        });
+        it('should pick red and yellow apples', function() {
```

Comment:
For that case you should also provide user with object or check if object is something you specified, because right know I can just hardcore all the properties I need.